File: buffercopy-py/buffercopy.py
import random
import time

assets = 1000000 # 200000

contentMetadataBuf = b'123456789a'
scoreBuf = b'bcde'

contentMetadataLen = len(contentMetadataBuf)
scoreBufLen = len(scoreBuf)
contentRespBufSize = contentMetadataLen + scoreBufLen

contentBufList = [contentMetadataBuf for i in range(assets)]
scoreBufList = [scoreBuf for i in range(assets)]

indexes = [random.randint(0, assets-1) for i in range(assets)]

for i in range(1000):
  startMs = int(round(time.time() * 1000))

  respBuf = bytearray(assets*contentRespBufSize)

  for i in range(assets):
    respBuf[i*contentRespBufSize:i*contentRespBufSize+contentMetadataLen] = contentBufList[indexes[i]]
    respBuf[i*contentRespBufSize+contentMetadataLen:i*contentRespBufSize+contentMetadataLen+scoreBufLen] = scoreBufList[indexes[i]]

  endMs = int(round(time.time() * 1000))
  print(endMs - startMs)

File: buffercopy-node/buffercopy.js
const assets = 1000000 //200000

const contentMetadataBuf = Buffer.allocUnsafe(10);
for (let i = 0; i < 10; i++)  contentMetadataBuf[i] = i+1

const scoreBuf = Buffer.allocUnsafe(4);
for (let i = 0; i < 4; i++) scoreBuf[i] = i+11

const contentMetadataLen = contentMetadataBuf.length
const scoreBufLen = scoreBuf.length
const contentRespBufSize = contentMetadataLen + scoreBufLen

const contentBufList = []
const scoreBufList = []
const indexes = []

for (let i = 0; i < assets; i++) {
    contentBufList[i] = contentMetadataBuf
    scoreBufList[i] = scoreBuf
    indexes[i] = Math.floor(Math.random() * Math.floor(assets))
}

for (let i = 0; i < 1000; i++) {
  const startMs = Date.now()

  const respBuf = Buffer.allocUnsafe(assets*contentRespBufSize)

  for (let i = 0; i < assets; i++) {
    contentBufList[indexes[i]].copy(respBuf, i*contentRespBufSize)
    scoreBufList[indexes[i]].copy(respBuf, i*contentRespBufSize+contentMetadataLen)
  }

  const endMs = Date.now()
  console.log(endMs - startMs)
}